packages/dcos-integration-test/extra/test_networking.py

```diff
@@ -28,85 +32,295 @@ def lb_enabled():
     r = requests.post(proxy_uri, data=cmd)
     log.info('Requests Response: %s', repr(r.json()))
     assert r.json()['status'] == 0
     return json.loads(r.json()['output'])
 
 
-class VipTest:
-    def __init__(self, num, vip, vipaddr, samehost, vipnet, proxynet):
-        self.vip = vip.format(num, 7000 + num)
-        self.vipaddr = vipaddr.format(num, 7000 + num)
+class VipTest():
+
+    def __init__(self, num, container, is_static, samehost, vipnet, proxynet, notes=''):
+        self.vip = '1.1.1.{}:{}'
+        self.vipaddr = self.vip
+        if is_static:
+            self.vip = '/namedvip{}:{}'
+            self.vipaddr = 'namedvip{}.marathon.l4lb.thisdcos.directory:{}'
+        self.vip = self.vip.format(num, 7000 + num)
+        self.vipaddr = self.vipaddr.format(num, 7000 + num)
+        self.num = num
+        self.container = container
         self.samehost = samehost
         self.vipnet = vipnet
         self.proxynet = proxynet
+        self.notes = notes
 
     def __str__(self):
-        return ("VipTest(vip={},vipaddr={},samehost={},"
-                "vipnet={},proxynet={})").format(self.vip, self.vipaddr, self.samehost, self.vipnet, self.proxynet)
+        s = 'VipTest(container={}, vip={},vipaddr={},samehost={},vipnet={},proxynet={}) {}'
+        return s.format(self.container, self.vip, self.vipaddr, self.samehost, self.vipnet, self.proxynet,
+                        self.notes)
 
     def log(self, s, lvl=logging.DEBUG):
         m = 'VIP_TEST {} {}'.format(s, self)
         log.log(lvl, m)
 
+    def docker_vip_app(self, network, host, vip):
+        app, uuid = get_test_app_in_docker()
+        app['id'] = '/viptest/' + app['id']
+        app['container']['docker']['network'] = network
+
+        app['mem'] = 16
+        app['cpu'] = 0.01
+        if network == 'HOST':
+            app['cmd'] = '/opt/mesosphere/bin/dcos-shell python '\
+                         '/opt/mesosphere/active/dcos-integration-test/util/python_test_server.py $PORT0'
+            del app['container']['docker']['portMappings']
+            if vip is not None:
+                app['portDefinitions'] = [{'labels': {'VIP_0': vip}}]
+
+        else:
+            app['cmd'] = '/opt/mesosphere/bin/dcos-shell python '\
+                         '/opt/mesosphere/active/dcos-integration-test/util/python_test_server.py 9080'
+            app['container']['docker']['portMappings'] = [{
+                'hostPort': 0,
+                'containerPort': 9080,
+                'protocol': 'tcp',
+                'name': 'test',
+                'labels': {}
+            }]
+            if vip is not None:
+                app['container']['docker']['portMappings'][0]['labels'] = {'VIP_0': vip}
+            if network == 'USER':
+                app['ipAddress'] = {'networkName': 'dcos'}
+
+        app['constraints'] = [['hostname', 'CLUSTER', host]]
+        return app, uuid
+
+    def mesos_vip_app(self, num, network, host, vip, ucr=False):
```

Comment:
Porting from the previous PR. Please use newline generously to improve code-readability.